scalyr_agent/copying_manager/common.py

```diff
@@ -16,27 +16,83 @@
 from __future__ import unicode_literals
 from __future__ import absolute_import
 
+import os
 
 if False:
     from typing import Dict
+    from typing import Optional
 
 from scalyr_agent import util as scalyr_util
+from scalyr_agent.scalyr_logging import AgentLogger
+from scalyr_agent.configuration import Configuration
 
 import six
 
 
-def write_checkpoint_state_to_file(checkpoints, file_path, current_time):
-    # type: (Dict, six.text_type, float) -> None
+def read_checkpoint_state_from_file(
```

Comment:
Can you add documentation to each of these methods about who is allowed to invoke them?  For example, only the worker process should read the file containing it's checkpoint.  Same with update and write below.

We need some place to document the overall checkpointing strategy.  Who owns what, what the main process does, what the worker processes do, etc.